src/main/java/org/traccar/protocol/Gt06ProtocolDecoder.java

```diff
@@ -222,9 +222,19 @@ public static boolean decodeGps(Position position, ByteBuf buf, boolean hasLengt
 
         position.set(Position.KEY_SATELLITES, BitUtil.to(buf.readUnsignedByte(), 4));
 
+        decodeLocation(position, buf, true);
+
+        return true;
+    }
+
+    private static void decodeLocation(Position position, ByteBuf buf, boolean withSpeed) {
```

Comment:
Instead of extracting a new method, can we just pass some boolean flags?

Reply:
Thanks for the review.

The `decodeGps` method would have to have one flag for each:

* date
* length
* altitude (the part in `decodeExtendedModular` uses that)
* satelites
* speed

Looks like too many switches to be honest. Another solution could be an adapter on top of the buffer but that is a bit more effort.

Anyway, your project, your rules, your call - if you would like to have those switches, let me know and I will refactor this pull request.

Reply:
Yep, let's do the switches. I think we just need 2 new arguments, not 3. But let's see.